cloudformation-telemetry/lambda_function.py

```diff
@@ -0,0 +1,148 @@
+from collections import defaultdict
+from re import T
+import boto3
+from datetime import datetime
+import os
+import json
+import requests
+import time
+from crhelper import CfnResource
+from abc import ABC, abstractmethod
+
+
+helper = CfnResource(json_logging=False, log_level='INFO', sleep_on_delete=30)
+
+@helper.create
+def create(event, context):
+    try:
+        T = telemetryFactory(event, context)
+        T.fetch_and_send_telemetry()
+    except Exception as e:
+        print(e)
+        return "Telemetry failed to sent for Create Stack"    
+    helper.Status = "SUCCESS"
+    return "Telemetry sent for Create Stack"
+
+
+@helper.update
+def update(event, context):
+    try:
+        T = telemetryFactory(event, context)
+        T.fetch_and_send_telemetry()
+    except Exception as e:
+        print(e)
+        return "Telemetry failed to sent for Update Stack"    
+    helper.Status = "SUCCESS"
+    return "Telemetry sent for Update Stack"
+
+
+@helper.delete
+def delete(event, context):
+    try:
+        T = telemetryFactory(event, context)
+        T.fetch_and_send_telemetry()
+    except Exception as e:
+        print(e)
+    helper.Status = "SUCCESS"
+
+
+def lambda_handler(event, context):
+    helper(event, context)
+
+def telemetryFactory(event, context):
+    if os.getenv['solutionName'] == 'AWSO':
+        return awsoTelemetry(event, context)
+    else:
+        return None
```

Comment:
here we can raise the exception instead of returning None. Otherwise it will throw None object does not have fetch_and_send_telemetry error